internal/pkg/deploy/pipeline.go

```diff
@@ -729,3 +741,15 @@ func (a *TestCommandsAction) Name() string {
 func (a *TestCommandsAction) Commands() []string {
 	return a.commands
 }
+
+// Action represents a CodePipeline action.
+type Action struct {
+	action
```

Comment:
Does this make more sense to you?
```go
// Step represents a CodePipeline action backed by a CodeBuild project.
type Step struct {
	action
	Build
	name string
}
```

I think that was because of a bad naming. The struct named `Action` was actually a specific "type" of action, it is not equivalent to the generic `action` struct.